Realm/RLMQueryUtil.mm

```diff
@@ -726,10 +724,55 @@ Query make_diacritic_insensitive_constraint(NSPredicateOperatorType operatorType
                                @"Operator '%@' not supported for string queries on Dictionary.",
                                operatorName(operatorType));
             }
+            else {
+                static_assert(AlwaysFalse<C>::value, "unsupported column type");
+            }
         }
     }
 }
 
+template <typename C, typename T>
+void QueryBuilder::do_add_diacritic_sensitive_string_constraint(NSPredicateOperatorType operatorType,
+                                                                NSComparisonPredicateOptions predicateOptions,
+                                                                C&& column, T&& value) {
+    bool caseSensitive = !(predicateOptions & NSCaseInsensitivePredicateOption);
+    Query condition = make_diacritic_sensitive_constraint(operatorType, caseSensitive, column, value);
+
+    // Queries on Mixed used to coerce Strings to Binary and vice-versa. Core
+    // no longer does this, but we can maintain compatibility by doing the
+    // coercion and checking both
+    if constexpr (is_any_v<C, Columns<Mixed>, Columns<Lst<Mixed>>, Columns<Set<Mixed>>, Columns<Dictionary>>) {
+        Mixed m = value;
+        if (!m.is_null()) {
```

Comment:
With this changes are we replacing some core removed implementation?, to maintain compatibility. Does that mean this is no longer needed, and we can remove this in major a version?

Reply:
Correct, this can go away with a major version bump. I'll add a comment noting that.